api/types/databaseobjectimportrule/databaseobjectimportrule_test.go

```diff
@@ -0,0 +1,172 @@
+				Spec: &dbobjectimportrulev1.DatabaseObjectImportRuleSpec{
+					DbLabels: map[string]string{"key": "value"},
+					Mappings: []*dbobjectimportrulev1.DatabaseObjectImportRuleMapping{{}},
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:          "nil rule",
+			rule:          nil,
+			expectedError: trace.BadParameter("must be non-nil"),
+		},
+		{
+			name: "missing metadata",
+			rule: &dbobjectimportrulev1.DatabaseObjectImportRule{
+				Kind:     types.KindDatabaseObjectImportRule,
+				Version:  types.V1,
+				Metadata: nil,
+				Spec:     &dbobjectimportrulev1.DatabaseObjectImportRuleSpec{},
+			},
+			expectedError: trace.BadParameter("metadata: must be non-nil"),
+		},
+		{
+			name: "missing name",
+			rule: &dbobjectimportrulev1.DatabaseObjectImportRule{
+				Kind:    types.KindDatabaseObjectImportRule,
+				Version: types.V1,
+				Metadata: &headerv1.Metadata{
+					Name:      "",
+					Namespace: defaults.Namespace,
+				},
+				Spec: &dbobjectimportrulev1.DatabaseObjectImportRuleSpec{},
+			},
+			expectedError: trace.BadParameter("metadata.name: must be non-empty"),
+		},
+		{
+			name: "invalid kind",
+			rule: &dbobjectimportrulev1.DatabaseObjectImportRule{
+				Kind:    "InvalidKind",
+				Version: types.V1,
+				Metadata: &headerv1.Metadata{
+					Name:      "test",
+					Namespace: defaults.Namespace,
+				},
+				Spec: &dbobjectimportrulev1.DatabaseObjectImportRuleSpec{},
+			},
+			expectedError: trace.BadParameter("wrong kind %v", "InvalidKind"),
+		},
+		{
+			name: "missing spec",
+			rule: &dbobjectimportrulev1.DatabaseObjectImportRule{
+				Kind:    types.KindDatabaseObjectImportRule,
+				Version: types.V1,
+				Metadata: &headerv1.Metadata{
+					Name:      "test",
+					Namespace: defaults.Namespace,
+				},
+				Spec: nil,
+			},
+			expectedError: trace.BadParameter("missing spec"),
+		},
+		{
+			name: "missing db_labels",
+			rule: &dbobjectimportrulev1.DatabaseObjectImportRule{
+				Kind:    types.KindDatabaseObjectImportRule,
+				Version: types.V1,
+				Metadata: &headerv1.Metadata{
+					Name:      "test",
+					Namespace: defaults.Namespace,
+				},
+				Spec: &dbobjectimportrulev1.DatabaseObjectImportRuleSpec{
+					Mappings: []*dbobjectimportrulev1.DatabaseObjectImportRuleMapping{{}},
+				},
+			},
+			expectedError: trace.BadParameter("missing db_labels"),
+		},
+		{
+			name: "missing mappings",
+			rule: &dbobjectimportrulev1.DatabaseObjectImportRule{
+				Kind:    types.KindDatabaseObjectImportRule,
+				Version: types.V1,
+				Metadata: &headerv1.Metadata{
+					Name:      "test",
+					Namespace: defaults.Namespace,
+				},
+				Spec: &dbobjectimportrulev1.DatabaseObjectImportRuleSpec{
+					DbLabels: map[string]string{"key": "value"},
+				},
+			},
+			expectedError: trace.BadParameter("missing mappings"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := ValidateDatabaseObjectImportRule(tt.rule)
+			require.ErrorIs(t, err, tt.expectedError)
```

Comment:
Please avoid ErrorIs. Same recommendation for ErrorContains and ErrorAs or IsBadParameter.